lib/util.js

```diff
@@ -36,83 +26,106 @@ const buildJSONresult = function(err, warn, inf, metadata) {
+                // Illegal params:
+                throw new Error('Illegal parameter “' + p + '”');
         }
-        else if ('validation' === p || 'htmlValidator' === p || 'cssValidator' === p ||
-            'patentPolicy' === p || 'processDocument' === p || 'noRecTrack' === p ||
-            'informativeOnly' === p || 'echidnaReady' === p || 'events' === p) {
-            // Other params:
-            if (result.hasOwnProperty(p))
-                throw new Error('Parameter “' + p + '” is used more than once');
-            result[p] = params[p];
-        }
-        else if (!constraints || !constraints.allowUnknownParams)
-            // Illegal params:
-            throw new Error('Illegal parameter “' + p + '”');
-    }
-    if (!originFound)
-        // Origin: one required.
-        throw new Error('One parameter of {“url”, “source”, “file”, “document”} is required');
-    else {
-        var c;
-        if(constraints && constraints.required) {
-            // Extra required params:
-            for (c in constraints.required)
-                if (!result.hasOwnProperty(constraints.required[c]))
-                    throw new Error('Parameter “' + constraints.required[c] + '” is required in this context');
-        }
-        if(constraints && constraints.forbidden) {
-            // Forbidden params:
-            for (c in constraints.forbidden)
-                if (result.hasOwnProperty(constraints.forbidden[c]))
-                    throw new Error('Parameter “' + constraints.forbidden[c] + '” is not allowed in this context');
+        if (!originFound)
+            // Origin: one required.
+            throw new Error('One parameter of {“url”, “source”, “file”, “document”} is required');
+        else {
+            var c;
+            if(constraints && constraints.required) {
+                // Extra required params:
+                for (c in constraints.required)
+                    if (!result.hasOwnProperty(constraints.required[c]))
+                        throw new Error('Parameter “' + constraints.required[c] + '” is required in this context');
+            }
+            if(constraints && constraints.forbidden) {
+                // Forbidden params:
+                for (c in constraints.forbidden)
+                    if (result.hasOwnProperty(constraints.forbidden[c]))
+                        throw new Error('Parameter “' + constraints.forbidden[c] + '” is not allowed in this context');
+            }
         }
-    }
-    return result;
+        return result;
+    };
 };
 
-exports.profiles = profiles;
-exports.buildJSONresult = buildJSONresult;
-exports.processParams = processParams;
+fs.readdir('./lib/profiles/', (err, files) => {
+    if (err)
+        throw new Error(`Cannot build profiles in dir "/lib/profiles/": "${err}"`);
+    else {
+        const profiles = {};
+        files.forEach((i) => {
+            const valid = /(^[A-Z][A-Z\-]*[A-Z](\-Echidna)?)\.js$/.exec(i);
```

Comment:
You're right: it's an ugly hack. I'll restore uppercases and move profiles under those two subdirs you suggest.